src/components/TrailForm.js

```diff
@@ -160,8 +178,45 @@ class TrailForm extends React.Component {
 
             </List>
 
+            <Button
+              fullWidth
+              style={{ marginTop: 10 }}
+              color="secondary"
+              variant="raised"
+              onClick={() => { this.toggleConfirmation(true)}}
+            > Delete Trail
+            </Button>
+
           </CardContent>
         </Card>
+
+        <Dialog open={dialogOpen} >
```

Comment:
This seems like a good candidate for a separate component. For bonus points, try making a generally re-purposable `ConfirmationDialog` component that could potentially be used in other places in the application (deleting a hydrant, confirming saving the data to the db later, etc).